### PR TITLE
Create tests for OrganizationDashboard.tsx

### LINKED ISSUE #1132: Code Coverage: Create tests for src/screens/OrganizationDashboard/OrganizationDashboard.tsx
We will need unittests done for all methods, classes and functions found in this file.

Any widgets, components, modals referenced in this file must also have unittests done

### PR Acceptance Criteria

- When complete this file must show at least **100%** coverage when merged into the code base. 
- [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin?search=&displayType=list)
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Sauradip07**
**Create tests for src/screens/OrganizationDashboard/OrganizationDashboard.tsx**

**Issue Number:**

Fixes #1132

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

Screenshot from 2024-02-06 15-20-51

Screenshot from 2024-02-06 15-21-11

**Summary**

Lines 86-94 & Lines 239-241 weren't covered by the tests It seems that Upcoming Events are mainly not covered
which introduce in Last PR of this File 

**Other information**

I try to cover the entire useEffect hook 
for Line 86 to 94 
```
  // UseEffect to update upcomingEvents array
  useEffect(() => {
    if (eventData && eventData?.eventsByOrganizationConnection.length > 0) {
      const tempUpcomingEvents: InterfaceQueryOrganizationEventListItem[] = [];
      eventData?.eventsByOrganizationConnection.map((event: any) => {
        const startDate = new Date(event.startDate);
        const now = new Date();
        if (startDate > now) {
          tempUpcomingEvents.push(event);
        }
      });
      setUpcomingEvents(tempUpcomingEvents);
    }
  }, [eventData?.eventsByOrganizationConnection]);
```
The 239 to 241 is already Covered but not showed in Codecov
**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

`Please feel free to reach out if there are any modifications or any changes need to be made.`

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@beingnoble03 @Kevoniat Can you review this PR?

**Comment by Sauradip07** [comment]
> can you share the test coverage percentage? @Sauradip07
Screenshot from 2024-02-06 22-13-52
Codecov report

File: src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx [hunk 1]
```diff
@@ -147,4 +148,78 @@ describe('Organisation Dashboard Page', () => {
...
+
+  test('Testing useEffect hook and error redirection', async () => {
+    const mockEventData = {
+      eventsByOrganizationConnection: [
+        { startDate: new Date().toISOString() }, // Assuming an event is upcoming
+      ],
+    };
+
+    // Mocking the response for the organization event query
+    const linkMocked = new StaticMockLink(
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:160]: Move this mock to `'./OrganizationDashboardMocks'` like other mocks.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
Please address the comments.

File: src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts [hunk 2]
```diff
@@ -298,3 +299,23 @@ export const ERROR_MOCKS = [
...
+    {
+      request: {
+        query: ORGANIZATION_EVENT_CONNECTION_LIST,
+        variables: { organization_id: 'your_organization_id' },
+      },
+      result: { data: mockEventData },
+    },
+  ],
+  true
+);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts:321]: No need to create `linkMocked` here, move 
```ts
    {
      request: {
        query: ORGANIZATION_EVENT_CONNECTION_LIST,
        variables: { organization_id: 'your_organization_id' },
      },
      result: { data: mockEventData },
    },
```
to `MOCKS`. Please follow the convention.

File: src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts [hunk 3]
```diff
@@ -298,3 +299,23 @@ export const ERROR_MOCKS = [
     error: new Error('Mock Graphql ORGANIZATION_EVENT_LIST Error'),
   },
 ];
+
+export const mockEventData = {
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts:303]: Remove `mockEventData` from here. We are using this only in a single test. Declare the variable inside the test itself (i.e. inside this test `Testing useEffect hook and error redirection`)

File: src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts (hunk 2 above)
Comment (beingnoble03) [reply src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts:321]: Use `link1` then.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
Please follow the convention.

File: src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts (hunk 2 above)
Comment (Sauradip07) [reply src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts]: I insert this code into the  `MOCKS`
  
```
    {
      request: {
        query: ORGANIZATION_EVENT_CONNECTION_LIST,
        variables: { organization_id: 'your_organization_id' },
      },
      result: { data: mockEventData },
    },
```
Please Check, is it right or wrong

File: src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts (hunk 3 above)
Comment (Sauradip07) [reply src/screens/OrganizationDashboard/OrganizationDashboardMocks.ts]: Done please Check

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by Sauradip07** [comment]
> Please fix the failing tests

The Test Failing because of `OrgList component`
Screenshot from 2024-02-11 22-18-54

In my PR, I don't change any file of `OrgList component`
The fix is Currently Done as a reference, I attach the SS here 
Screenshot from 2024-02-11 22-39-18

How I have to just re-run the CI to check is every thing is ok or not 
I guess that fix the Failing test
Should I push Empty commit to run the CI

**Comment by palisadoes** [comment]
This should have been fixed. Do a minor commit to trigger the workflow

**Comment by palisadoes** [comment]
You added too many new files.

1. Please fix the conflicting files. They must all refer to our repo not yours. 
2. Merge with the latest upstream before

**Pull request closed without merging**